src/App.tsx

```diff
@@ -15,48 +16,118 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type State = {
+  isReversed: boolean,
+  sortType: SortType,
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { isReversed, sortType }: State,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((g1, g2) => {
+    switch (sortType) {
+      case SortType.LENGTH:
+        return g1.length - g2.length;
+
+      case SortType.ALPHABET:
+        return g1.localeCompare(g2);
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [isReversed, setReversed] = useState(false);
+  const [sortType, setType] = useState(SortType.NONE);
+
+  const onReset = () => {
+    setType(SortType.NONE);
+    setReversed(false);
+  };
+
+  const sortByAlphabet = (typeOfSort: SortType) => () => setType(typeOfSort);
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={classNames(
+            'button',
+            'is-info',
+            {
+              'is-light': sortType !== SortType.ALPHABET,
+            },
+          )}
+          // onClick={this.sortByAlphabet}
```

Comment:
Remove comment.

Reply:
I fixed this part of the code. Thank you!